static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -0,0 +1,463 @@
+import targetingTool from 'common/modules/commercial/targeting-tool';
+import {
+    regulars as acquisitionsCopyRegulars,
+    control as acquisitionsCopyControl,
+} from 'common/modules/commercial/acquisitions-copy';
+import type { AcquisitionsEpicTemplateCopy } from 'common/modules/commercial/acquisitions-copy';
+import { control as acquisitionsTestimonialParametersControl } from 'common/modules/commercial/acquisitions-epic-testimonial-parameters';
+import type { AcquisitionsEpicTestimonialTemplateParameters } from 'common/modules/commercial/acquisitions-epic-testimonial-parameters';
+import { logView } from 'common/modules/commercial/acquisitions-view-log';
+import { isRegular } from 'common/modules/tailor/tailor';
+import $ from 'lib/$';
+import config from 'lib/config';
+import { getCookie } from 'lib/cookies';
+import ElementInView from 'lib/element-inview';
+import fastdom from 'lib/fastdom-promise';
+import mediator from 'lib/mediator';
+import { getSync as geolocationGetSync } from 'lib/geolocation';
+import { constructQuery as constructURLQuery } from 'lib/url';
+import { noop } from 'lib/noop';
+import { daysSince } from 'lib/time-utils';
+import template from 'lodash/utilities/template';
+import toArray from 'lodash/collections/toArray';
+import acquisitionsEpicControlTemplate from 'raw-loader!common/views/acquisitions-epic-control.html';
+import acquisitionsTestimonialBlockTemplate from 'raw-loader!common/views/acquisitions-epic-testimonial-block.html';
+
+type ContributionsABTest = ABTest & {
+    epic: boolean,
+    campaignId: string,
+    campaignPrefix: string,
+    campaignSuffix: string,
+    useLocalViewLog: boolean,
+    overrideCanRun: boolean,
+    showToContributorsAndSupporters: boolean,
+    pageCheck: (page: Object) => boolean,
+    locations: Array<string>,
+    locationCheck: (location: string) => boolean,
+    useTargetingTool: boolean,
+    insertEvent: string,
+    viewEvent: string,
+};
+
+type EpicTemplate = (Variant, AcquisitionsEpicTemplateCopy) => string;
+
+const membershipBaseURL = 'https://membership.theguardian.com/supporter';
+const contributionsBaseURL = 'https://contribute.theguardian.com';
+
+const lastContributionDate = getCookie('gu.contributions.contrib-timestamp');
+
+/**
+ * How many times the user can see the Epic, e.g. 6 times within 7 days with minimum of 1 day in between views.
+ * @type {{days: number, count: number, minDaysBetweenViews: number}}
+ */
+const defaultMaxViews = {
+    days: 30,
+    count: 4,
+    minDaysBetweenViews: 0,
+};
+
+const controlTemplate: EpicTemplate = (variant, copy) =>
+    template(acquisitionsEpicControlTemplate, {
+        copy,
+        membershipUrl: variant.options && variant.options.membershipURL,
+        contributionUrl: variant.options && variant.options.contributeURL,
+        componentName: variant.options && variant.options.componentName,
+        testimonialBlock: variant.options && variant.options.testimonialBlock,
+    });
```

Comment:
We could at least destruct the `options` property out of `variant` in the params:
```js
const controlTemplate: EpicTemplate = ({ options }, copy) =>
```